Source/WebCore/platform/graphics/texmap/TextureMapper.h

```diff
@@ -65,11 +64,11 @@ class TextureMapper {
     void drawNumber(int number, const Color&, const FloatPoint&, const TransformationMatrix&);
 
     WEBCORE_EXPORT void drawTexture(const BitmapTexture&, const FloatRect& target, const TransformationMatrix& modelViewMatrix = TransformationMatrix(), float opacity = 1.0f, AllEdgesExposed = AllEdgesExposed::Yes);
-    void drawTexture(GLuint texture, OptionSet<TextureMapperFlags>, const FloatRect& targetRect, const TransformationMatrix& modelViewMatrix, float opacity, AllEdgesExposed = AllEdgesExposed::Yes);
-    void drawTexturePlanarYUV(const std::array<GLuint, 3>& textures, const std::array<GLfloat, 16>& yuvToRgbMatrix, OptionSet<TextureMapperFlags>, const FloatRect& targetRect, const TransformationMatrix& modelViewMatrix, float opacity, std::optional<GLuint> alphaPlane, AllEdgesExposed = AllEdgesExposed::Yes);
-    void drawTextureSemiPlanarYUV(const std::array<GLuint, 2>& textures, bool uvReversed, const std::array<GLfloat, 16>& yuvToRgbMatrix, OptionSet<TextureMapperFlags>, const FloatRect& targetRect, const TransformationMatrix& modelViewMatrix, float opacity, AllEdgesExposed = AllEdgesExposed::Yes);
-    void drawTexturePackedYUV(GLuint texture, const std::array<GLfloat, 16>& yuvToRgbMatrix, OptionSet<TextureMapperFlags>, const FloatRect& targetRect, const TransformationMatrix& modelViewMatrix, float opacity, AllEdgesExposed = AllEdgesExposed::Yes);
-    void drawTextureExternalOES(GLuint texture, OptionSet<TextureMapperFlags>, const FloatRect&, const TransformationMatrix& modelViewMatrix, float opacity);
+    void drawTexture(unsigned texture, OptionSet<TextureMapperFlags>, const FloatRect& targetRect, const TransformationMatrix& modelViewMatrix, float opacity, AllEdgesExposed = AllEdgesExposed::Yes);
+    void drawTexturePlanarYUV(const std::array<unsigned, 3>& textures, const std::array<float, 16>& yuvToRgbMatrix, OptionSet<TextureMapperFlags>, const FloatRect& targetRect, const TransformationMatrix& modelViewMatrix, float opacity, std::optional<unsigned> alphaPlane, AllEdgesExposed = AllEdgesExposed::Yes);
+    void drawTextureSemiPlanarYUV(const std::array<unsigned, 2>& textures, bool uvReversed, const std::array<float, 16>& yuvToRgbMatrix, OptionSet<TextureMapperFlags>, const FloatRect& targetRect, const TransformationMatrix& modelViewMatrix, float opacity, AllEdgesExposed = AllEdgesExposed::Yes);
+    void drawTexturePackedYUV(unsigned texture, const std::array<float, 16>& yuvToRgbMatrix, OptionSet<TextureMapperFlags>, const FloatRect& targetRect, const TransformationMatrix& modelViewMatrix, float opacity, AllEdgesExposed = AllEdgesExposed::Yes);
+    void drawTextureExternalOES(unsigned texture, OptionSet<TextureMapperFlags>, const FloatRect&, const TransformationMatrix& modelViewMatrix, float opacity);
```

Comment:
This change doesn't look acceptable.
You need pimpl idiom or restoring TextureMapper abstact class.

Reply:
Why? I don't see the problem. Note that this is working for you because you use angle for everything.

Reply:
Let's use `GraphicsTypesGL.h` like the rest of WebCore does, then.

Reply:
> GraphicsTypesGL.h is for GraphicsContextGL. 

Well, it's used in other places, in some cases we even include GraphicsContextGL.h when we really just need GraphicsTypesGL.h. But the idea is the same in any case, it uses its own types matching the GL ones to avoid including the GL headers.

> TextureMaper is using system OpenGL while GraphicsContextGL (WebGL) is using ANGLE in GTK and WPE ports.

Yes, and that's still the case, the goal of this patch is not to change that, only to avoid including the GL headers from the header classes.

> The previous is better design before 270048@main merged TextureMaperGL in order to hide OpenGL headers.

That was not the intention of the previous design, and it didn'0t work either, because TextureMapperGL.h already indluded the GL header and was included directly in some other files. The previous design comes from the times when it was possible to use TextureMapper with a software based path. That was removed and we kept the abstraction that was no longer needed.

>  https://bugs.webkit.org/show_bug.cgi?id=263961 We should use pimpl idiom or TextureMaper abstruct class to hide OpenGL headers.

I don't understand why, and I don't see how adding back the abstract class helps.

Reply:
> because TextureMapperGL.h already indluded the GL header and was included directly in some other files.

A callsite doesn't need to include TextureMapperGL.h. It should use `TextureMapper::create` factory class method instead.

> I don't understand why, and I don't see how adding back the abstract class helps.

abstract class can be considered as pimpl idiom.
I'm suggesting pimp idiom if you don't like the idea of abstract class.

Reply:
That was probably the initial intention, but now it's used in other parts not related to WebGL. Anyway, let's go back to the previous version and use int, unsigned, float. I still don't understand the problem of using those types when it's documented that GLint is int, GLuint unsigned int, etc.

Reply:
Other parts? Which code?
Your design looks pecuriler. The previous TextureMapperGL is a better design.
WebKit project is fastidious to even a single whitespace.

Reply:
No one says there is a problem in both designs. Your design looks pecuriler. We should move to webkit-dev.